sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/UnitTestHelper.cs

```diff
@@ -95,5 +95,57 @@ public static bool IsMsTestOrNUnitTestIgnored(this IMethodSymbol method)
                     a.AttributeClass.Is(KnownType.Xunit_FactAttribute) ||
                     a.AttributeClass.Is(KnownType.Xunit_TheoryAttribute) ||
                     a.AttributeClass.Is(KnownType.LegacyXunit_TheoryAttribute));
+
+        public static KnownType FindTestAttribute(this IMethodSymbol method)
```

Comment:
This is effective the old _ToKnownTestType_ method from the test, but expanded to cover all of the known test method types.

I don't like having a big lookup method like this one, but couldn't think of a simpler way of translating from attribute symbols to _KnownTypes_.